featuretools/primitives/utils.py

```diff
@@ -274,20 +251,15 @@ def _get_summary_primitives(primitives: List) -> Dict[str, int]:
 
 
 def _check_input_types(
-    input_types: List[ColumnSchema],
-    input_checks: set,
-    unique_input_types: set,
+    input_types: List[ColumnSchema], input_checks: set, unique_input_types: set
 ):
-    """Recursively checks for unique input types for a primitive."""
+    """Checks if any logical types or time indices occur in a list of Woodwork input types and keeps track of unique input types."""
     for in_type in input_types:
-        if isinstance(in_type, list):
-            _check_input_types(in_type, input_checks, unique_input_types)
-        else:
-            if "time_index" in in_type.semantic_tags:
-                input_checks.add("semantic_time_idx")
-            if in_type.logical_type:
-                input_checks.add(str(in_type.logical_type.__class__))
-            unique_input_types.add(str(in_type))
+        if "time_index" in in_type.semantic_tags:
+            input_checks.add("ct_semantic_time_idx")
+        if in_type.logical_type:
+            input_checks.add(str(in_type.logical_type.__class__))
```

Comment:
Pretty sure you could use `in_type.logical_type.type_string` here if you change line 215 to 

```
        log_type: 0 for log_type in list(list_logical_types()["type_string"])
```

There's definitely not much difference between the two, but I feel like it's nice to avoid using the dunder methods if we can.

Reply:
ahh good call, didn't realize that property was avaliable. definitely better to avoid dunder where possible.